File: part3/script/ruby/hello_client.rb
#!/usr/bin/env ruby

require 'thrift'
$:.push('gen-rb')
require 'hello_svc'

begin
  transport = Thrift::BufferedTransport.new(Thrift::Socket.new('localhost', 9095))
  protocol = Thrift::BinaryProtocol.new(transport)
  client = HelloSvc::Client.new(protocol)

  transport.open()

  res = client.getMessage('world')
  puts 'Message from server: ' + res
  transport.close()

rescue Thrift::Exception => tx
  print 'Thrift::Exception: ', tx.message, "\n"
end

File: part3/script/perl/HelloClient.pl
#!/usr/bin/env perl

use lib './gen-perl';

use Thrift::Socket;
use Thrift::BufferedTransport;
use Thrift::BinaryProtocol;
use helloSvc;

my $trans_ep    = Thrift::Socket->new("localhost", 9095);
my $trans_buf = Thrift::BufferedTransport->new($trans_ep);
my $proto  = Thrift::BinaryProtocol->new($trans_buf);
my $client    = helloSvcClient->new($proto);

$trans_ep->open;
my $msg = $client->getMessage("World!");
$trans_ep->close;

print $msg."\n"
